### PR TITLE
Run our test validation on PRs

### LINKED ISSUE #669: Create workflow for our test validation
The test validation we've added should run on each PR.

We want to ensure that each PR that might change any sample code still runs.

This should follow the workflows used for the other actions.

Open question: Should it be required? (I'm not sure how far we've gotten in validating all the existing samples)

### PR DESCRIPTION
**Pull request by BillWagner**
The test extractor / test validator should run on all PRs once this is merged.

Fixes #669

### CODE REVIEW

File: .github/workflows/test-examples.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      - '/standard/*.md'
+  workflow_dispatch:
+    inputs:
+      reason:
+        description: 'The reason for running the workflow'
+        required: true
+        default: 'Manual run'
+
+jobs:
+  word-converter:
```
Comment (jskeet) [inline .github/workflows/test-examples.yaml]: This isn't a word converter :)

**Review by jskeet — APPROVED** [review]
Let's change the job name, but otherwise fine.
(Maybe update ubuntu version across the board at some point?)

File: .github/workflows/test-examples.yaml (hunk 1 above)
Comment (BillWagner) [inline .github/workflows/test-examples.yaml]:
```suggestion
  test-extraction-and-runner
```

**Comment by BillWagner** [comment]
closing and reopening for a fresh build

**Pull request merged**